core/shared/src/main/scala/fs2/CompositeFailure.scala

```diff
@@ -0,0 +1,10 @@
+package fs2
+
+/**
+  * Certain implementation in fs2 may cause multiple failures from the different part of the code to be thrown and collected.
+  * In these scenarios, COmposite failure assures that user won't lose any intermediate failures.
+  */
+class CompositeFailure(
+  first: Throwable
+  , others: List[Throwable]
+) extends Throwable(s"Resources failed to finalize: [${first.getClass.getName}: ${first.getMessage}] + ${others.size} others", first)
```

Comment:
Recommend reorganizing this message to the "+ N others" part of the string shows up in the prefix and is hence unambiguous as to whether it is part of the first.getMessage. Something like: `if (exceptions.tail.headOption.isEmpty) exceptions.head.getMessage else s"Multiple exceptions were thrown (${exceptions.size}), first: ${exceptions.head.getMessage}"`

Reply:
ok